File: _workspace/02_meso_notes/split.js
var fso = new ActiveXObject("Scripting.FileSystemObject");
var ForReading = 1, ForWriting = 2;
var srcPath = "c:\\Users\\pn-t-kojima\\Documents\\novel-write\\_workspace\\02_meso_notes\\chapter_summaries_draft.md";

// Open text file as ADODB.Stream to support UTF-8
var inStream = new ActiveXObject("ADODB.Stream");
inStream.Type = 2; // Text
inStream.Charset = "utf-8";
inStream.Open();
inStream.LoadFromFile(srcPath);
var content = inStream.ReadText();
inStream.Close();

var lines = content.split('\n');
var inYaml = false;
var currentLines = [];
var chapterNum = 0;

for (var i = 0; i < lines.length; i++) {
    var line = lines[i].replace(/\r$/, '');
    var match = line.match(/^## 第(\d+)話/);
    if (match) {
        chapterNum = parseInt(match[1], 10);
    } else if (line.match(/^```yaml/)) {
        inYaml = true;
        currentLines = [];
    } else if (line.match(/^```/) && inYaml) {
        inYaml = false;
        if (chapterNum > 0) {
            var numStr = chapterNum < 10 ? "0" + chapterNum : "" + chapterNum;
            var filename = "chapter_" + numStr + "_summary.yaml";
            var outPath = "c:\\Users\\pn-t-kojima\\Documents\\novel-write\\04_meso_plot\\arc_01\\" + filename;
            
            var outStream = new ActiveXObject("ADODB.Stream");
            outStream.Type = 2;
            outStream.Charset = "utf-8";
            outStream.Open();
            outStream.WriteText(currentLines.join('\n'));
            outStream.SaveToFile(outPath, 2); // adSaveCreateOverWrite
            outStream.Close();
        }
    } else if (inYaml) {
        currentLines.push(line);
    }
}

File: _workspace/archive/02_meso_notes/split.py
import os
import re

draft_path = r"c:\Users\pn-t-kojima\Documents\novel-write\_workspace\02_meso_notes\chapter_summaries_draft.md"
out_dir = r"c:\Users\pn-t-kojima\Documents\novel-write\04_meso_plot\arc_01"

with open(draft_path, "r", encoding="utf-8") as f:
    content = f.read()

# Find all yaml blocks
pattern = re.compile(r'## 第(\d+)話.*?```yaml\n(.*?)\n```', re.DOTALL)
matches = pattern.findall(content)

for num_str, yaml_content in matches:
    num = int(num_str)
    filename = f"chapter_{num:02d}_summary.yaml"
    filepath = os.path.join(out_dir, filename)
    with open(filepath, "w", encoding="utf-8") as f:
        f.write(yaml_content.strip() + "\n")
    print(f"Created {filename}")
